controllers/authpolicy_authconfig.go

```diff
@@ -174,6 +183,35 @@ func (r *AuthPolicyReconciler) desiredAuthConfig(ctx context.Context, ap *api.Au
 	return mergeConditionsFromRouteSelectorsIntoConfigs(ap, route, authConfig)
 }
 
+// routeGatewayHasAuthOverrides return true when the gateway which a route is attached to has an attached authPolicy that defines atomic overrides
+func routeGatewayHasAuthOverrides(ctx context.Context, route *gatewayapiv1.HTTPRoute, c client.Client) (bool, error) {
+	for idx := range route.Spec.ParentRefs {
+		parentRef := route.Spec.ParentRefs[idx]
+		gw := &gatewayapiv1.Gateway{}
+		err := c.Get(ctx, client.ObjectKey{Name: string(parentRef.Name), Namespace: string(*parentRef.Namespace)}, gw)
+		if err != nil {
+			return false, err
+		}
+
+		annotation, ok := gw.GetAnnotations()[common.AuthPolicyBackRefAnnotation]
+		if !ok {
+			continue
+		}
+		anno := strings.Split(annotation, "/")
+		ap := &api.AuthPolicy{}
+		err = c.Get(ctx, client.ObjectKey{Name: anno[1], Namespace: anno[0]}, ap)
+		if err != nil {
+			return false, err
+		}
+
+		override := ap.IsAtomicOverride()
+		if override {
+			return true, nil
+		}
```

Comment:
nit: unnessary if block ?
```suggestion
		return ap.IsAtomicOverride(), nil
```

Reply:
I suppose you want to continue in the loop until the end, because this parent may not have an override, but the next one yes?

Reply:
Oh, right, yes, I missed that, please disregard this comment